verification.py

```diff
@@ -878,6 +936,192 @@ def test_sweep_tau_ex3(self):
+            for tree in rts.trees():
+                t_mrca[tree.index] = tree.time(tree.root)
+
+            data["tmrca_mean"].append(np.mean(t_mrca))
+            data["num_trees"].append(rts.num_trees)
+            slim_win_pis.append(rts.diversity(windows=wins))
+            data["pi"].append(rts.diversity().reshape((1,))[0])
+            data["model"].append("slim")
+        df = df.append(pd.DataFrame(data))
+
+        df_slim = df[df.model == "slim"]
+        df_msp = df[df.model == "msp"]
+        for stat in ["tmrca_mean", "num_trees", "pi"]:
+            v1 = df_slim[stat]
+            v2 = df_msp[stat]
+            sm.graphics.qqplot(v1)
+            sm.qqplot_2samples(v1, v2, line="45")
+            pyplot.xlabel("msp")
+            pyplot.ylabel("SLiM")
+            f = self.output_dir / f"{stat}.png"
+            pyplot.savefig(f, dpi=72)
+            pyplot.close("all")
+            plot_stat_hist(v1, v2, v1.name, v2.name)
+            f = self.output_dir / f"{stat}.hist.png"
+            pyplot.savefig(f, dpi=72)
+            pyplot.close("all")
+        pyplot.plot(mids, np.array(msp_win_pis).mean(axis=0), label="msp")
+        pyplot.plot(mids, np.array(slim_win_pis).mean(axis=0), label="slim")
+        pyplot.title(f"tau: {tau}")
+        pyplot.legend()
+        f = self.output_dir / "pi_wins.png"
+        pyplot.savefig(f, dpi=72)
+        pyplot.close("all")
+
+    def _run(
+        self,
+        sample_size,
+        seq_length,
+        pop_size,
+        recombination_rate,
+        s,
+        tau,
+        num_replicates=None,
+    ):
+        """
+        basic tests for sweeps vs slim
+        """
+        slim_args = {}
+
+        if num_replicates is None:
+            num_replicates = 20
+
+        # These are *diploid* samples in msprime
+        slim_args["sample_size"] = 2 * sample_size
+        slim_args["r"] = recombination_rate
+        slim_args["NUMLOCI"] = int(seq_length - 1)
+        slim_args["POPSIZE"] = int(pop_size)
+        slim_args["TAU"] = tau
+        slim_args["s"] = s
+        slim_args["SWEEPPOS"] = int(seq_length / 2)
+        self.run_sweep_slim_comparison(
+            slim_args,
+            pop_size=pop_size,
+            sample_size=sample_size,
+            num_replicates=num_replicates,
+            seq_length=seq_length,
+            tau=tau,
+            s=s,
+            recombination_rate=recombination_rate,
+        )
+
+    def test_sweep_vs_slim_ex1(self):
+        self._run(10, 1e6, 1e3, 1e-7, 0.25, 1, num_replicates=100)
+
+    def test_sweep_vs_slim_ex2(self):
+        self._run(10, 1e6, 1e3, 1e-7, 0.25, 200, num_replicates=100)
+
+    def test_sweep_vs_slim_ex3(self):
+        self._run(10, 1e6, 1e3, 1e-7, 0.25, 1000, num_replicates=100)
+
+    def test_sweep_vs_slim_ex4(self):
+        self._run(10, 1e6, 1e3, 1e-7, 0.25, 2000, num_replicates=100)
+
+    def test_sweep_vs_slim_ex5(self):
+        self._run(10, 1e6, 1e3, 1e-7, 0.25, 5000, num_replicates=100)
```

Comment:
On my desktop, it took about 20 minutes to run the five tests. Maybe we can change the parameters a bit to make it run faster?

Reply:
i reduced this to 10 reps but it will be quite noisy now...